### PR TITLE
Fix diassociate EE from JT and WFJT 

### PR DESCRIPTION
**Pull request by nixocio**
Allow EE to be removed from JT and WFJT.

Also, add unit-test related to those changes.

See: https://github.com/ansible/awx/issues/9487

### CODE REVIEW

**Comment by nixocio** [comment]
Running e2e tests.

File: awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx [hunk 1]
```diff
@@ -266,6 +287,8 @@ describe('<JobTemplateEdit />', () => {
         id: 1,
         organization: 1,
       });
+
+      wrapper.find('ExecutionEnvironmentLookup').invoke('onChange')(null);
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx:291]: :heavy_check_mark:

**Review by jakemcdermott — APPROVED** [review]

**Comment by tiagodread** [comment]
@nixocio @jakemcdermott the fix worked well but there’s another thing on the details view after Saving the form, it removes the EE after some period, see:

jt3

File: awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx [hunk 2]
```diff
@@ -277,6 +300,7 @@ describe('<JobTemplateEdit />', () => {
       ...mockJobTemplate,
       project: mockJobTemplate.project,
       ...updatedTemplateData,
+      execution_environment: null,
     };
     delete expected.summary_fields;
     delete expected.id;
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx:306]: @nixocio I think we'll need to do something similar for WFJTs

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

**Comment by nixocio** [comment]
> @nixocio @jakemcdermott the fix worked well but there’s another thing on the details view after Saving the form, it removes the EE after some period, see:
> 
> jt3

This issue was not introduced by this PR. Please, try to edit the project, even in devel - and similar issue will be shown. I can attempt to fix this behavior here, but that seems to be a different issue - Not related to the disassociate EE.

File: awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx (hunk 2 above)
Comment (nixocio) [reply awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx:306]: @jakemcdermott, updated. Good catch.  Running e2e tests once more.

**Comment by tiagodread** [comment]
Opened #9516 to address this as a separate issue

**Review by tiagodread — APPROVED** [review]

File: awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx (hunk 2 above)
Comment (nixocio) [reply awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx:306]: e2e tests were 🟢

**Review by jakemcdermott — APPROVED** [review]

**Pull request merged**